--- cpp/code/handmade_renderer_opengl.cpp	2018-09-06 00:42:18.000000000 -0400
+++ handmade_renderer_opengl.cpp	2018-09-06 00:42:15.000000000 -0400
@@ -1250,7 +1250,8 @@
     DepthMax = max(DepthMax, Depth);
     }
     
-    gl_FragDepth = 0.5f*(DepthMin+DepthMax);
+    //gl_FragDepth = 0.5f*(DepthMin+DepthMax);
+	gl_FragDepth = DepthMax;
     
     v4 CombinedColor = V4(0, 0, 0, 0);
     v4 CombinedEmit = V4(0, 0, 0, 0);
@@ -2277,7 +2278,8 @@
                              GL_STREAM_DRAW);
                 
                 zbias_program *Prog = &OpenGL.ZBiasNoDepthPeel;
-                f32 AlphaThreshold = 0.0f;
+                //f32 AlphaThreshold = 0.0f;
+                f32 AlphaThreshold = 0.02f;
                 if(Peeling)
                 {
                     opengl_framebuffer *PeelBuffer = GetDepthPeelReadBuffer(OnPeelIndex - 1);
